source/data-modeling/validation.txt

```diff
@@ -0,0 +1,344 @@
+   queries a secondary member of the replica set, there is a possibility
+   that it is reading stale data.
+
+   This method takes a ``conditions`` option that allows you to specify
+   condition to add when {+odm+} checks for uniqueness:
+
+   .. code-block:: ruby
+
+      validates_uniqueness_of :name, conditions: -> { where(:age.gte => 10) }
+
+.. _mongoid-association-validation:
+
+Validate Associations
+~~~~~~~~~~~~~~~~~~~~~
+
+You can use the ``validates_associated`` helper to validate any
+associations that your model has. When you include this validation rule,
+{+odm+} validates any association documents any time you try to
+save an instance.
+
+.. TODO link to associations page
+
+This example defines an association validation rule on the
+``Author`` model to run the validation rules for the embedded ``Book``
+instances:
+
+.. literalinclude:: /includes/data-modeling/validation.rb
+   :start-after: start-assoc
+   :end-before: end-assoc
+   :language: ruby
+   :emphasize-lines: 6
+   :dedent:
+
+.. important::
+
+   Don't use the ``validates_associated`` helper on both ends of your
+   associations, as this causes {+odm+} to perform validations in an
+   infinite loop.
+
+.. _mongoid-other-validation:
+
+Custom Validation Rules
+~~~~~~~~~~~~~~~~~~~~~~~
+
+You can use the ``validates_each`` and ``validates_with`` helpers to
+create custom validators. To learn more about these helpers and view
+examples, see the `validates_each
+<https://guides.rubyonrails.org/active_record_validations.html#validates-each>`__
+and `validates_with
+<https://guides.rubyonrails.org/active_record_validations.html#validates-with>`__
+references in the Active Record documentation.
+
+To learn more about custom validators, see `Performing Custom
+Validations
+<https://guides.rubyonrails.org/active_record_validations.html#performing-custom-validations>`__
+in the Active Record documentation.
+
+Behavior
+--------
+
+{+odm+} performs validation when you persist, or save, a document to the
+database. The following methods trigger your validation rules, so
+{+odm+} saves the object to the database only if it passes validation:
+
+- ``create()``
+- ``save()``
+- ``update()``
+
+When you use the ``-!`` suffixed version of the preceding methods,
+{+odm+} returns an ``Mongoid::Errors::Validations`` exception if
+validation fails for an object.
+
+Trigger Validation
+~~~~~~~~~~~~~~~~~~
+
+You can run validations manually by using the ``valid?()`` method.This
+method returns ``true`` if the object passes validation, and
+``false`` otherwise:
+
+.. literalinclude:: /includes/data-modeling/validation.rb
+   :start-after: start-valid
+   :end-before: end-valid
+   :language: ruby
+   :emphasize-lines: 7, 11, 14
+   :dedent:
+
+{+odm+} behaves differently to Active Record when running ``valid?()``
+on persisted data. Active Record's ``valid?()`` runs all
+validations, but {+odm+}'s ``valid?()`` runs validations only on
+documents that are in memory as an optimization.
```

Comment:
[q] I wasn't sure if this sentence is saying that the way Mongoid runs validations is the optimization, or the documents are in memory as optimization? Maybe add a comma for clarity, or omit: 
```suggestion
documents that are in memory, as an optimization.
```
```suggestion
documents that are in memory.
```

Reply:
improved wording